x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -731,12 +741,25 @@ private void executeTask(DataFrameAnalyticsTaskState analyticsTaskState, Allocat
             }
         }
 
-        public static String nodeFilter(DiscoveryNode node, String id) {
+        public static String nodeFilter(DiscoveryNode node, TaskParams params) {
+            String id = params.getId();
 
-            if (node.getVersion().before(StartDataFrameAnalyticsAction.TaskParams.VERSION_INTRODUCED)) {
+            if (node.getVersion().before(TaskParams.VERSION_INTRODUCED)) {
                 return "Not opening job [" + id + "] on node [" + JobNodeSelector.nodeNameAndVersion(node)
                     + "], because the data frame analytics requires a node of version ["
-                    + StartDataFrameAnalyticsAction.TaskParams.VERSION_INTRODUCED + "] or higher";
+                    + TaskParams.VERSION_INTRODUCED + "] or higher";
+            }
+            if (node.getVersion().onOrAfter(TaskParams.VERSION_DESTINATION_INDEX_MAPPINGS_CHANGED)
```

Comment:
We should support rolling upgrades for data frame analytics jobs.

The experience here would be really bad. The user kicks off a rolling upgrade and the job basically stays in unassigned state forever. 

We should check the job version and if it is less than 7.10, then we need to set the state back before reindex, and start from the reindex state again.

Reply:
> We should support rolling upgrades for data frame analytics jobs

We said we wouldn't put the effort into that while it's experimental.

> The experience here would be really bad. The user kicks off a rolling upgrade and the job basically stays in unassigned state forever.

True.  We should probably add logic so that it an assignment is attempted and _no_ node in the cluster has the required version then the job gets stopped.  Then the user can restart it and it will go through the logic of https://github.com/elastic/elasticsearch/pull/62960.

Reply:
#62960 deletes the destination index if it is too old. Which covers this branch. 

This predicate should be removed.

Reply:
Done.